tensorflow/core/kernels/crop_resize_bilinear_core.h

```diff
@@ -0,0 +1,5651 @@
+/* Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_CORE_KERNELS_CROP_RESIZE_BILINEAR_CORE_H_
+#define TENSORFLOW_CORE_KERNELS_CROP_RESIZE_BILINEAR_CORE_H_
+
+#include <emmintrin.h>
+#include <immintrin.h>
+#include <smmintrin.h>
+#include <xmmintrin.h>
+#include <cassert>
+#include <cstdio>
+#include <cstdlib>
+#include <list>
+#include <string>
+
+namespace tensorflow {
+namespace {
+
+template <typename T>
+std::string sprint_bits(const T& val, bool msb_left = true) {
+  std::string str;
+  if (sizeof(T) == 1) {
+    unsigned char* p = (unsigned char*)&val;
```

Comment:
Use reinterpret_cast<unsigned char*>(&val) here instead of C-style cast (same below)